code/game/objects/structures/plaques/_plaques.dm

```diff
@@ -0,0 +1,171 @@
+			return
+		to_chat(user, "<span class='warning'>Your pen isn't fancy enough to engrave this! Find a fountain pen.</span>") //Go steal the Curator's.
+		return
+	return ..()
+
+/obj/item/plaque/attackby(obj/item/I, mob/user, params) //Same as part of the above, except for the item in hand instead of the structure.
+	if(istype(I, /obj/item/pen/fountain))
+		if(engraved)
+			to_chat(user, "<span class='warning'>This plaque has already been engraved.</span>")
+			return
+		var/namechoice = input(user, "Title this plaque. (e.g. 'Best HoP Award', 'Great Ashwalker War Memorial')", "Plaque Customization")
+		if(!namechoice)
+			return
+		var/descriptionchoice = input(user, "Engrave this plaque's text.", "Plaque Customization")
+		if(!descriptionchoice)
+			return
+		if(!Adjacent(user)) //Make sure user is adjacent still
+			to_chat(user, "<span class='warning'>You need to stand next to the plaque to engrave it!</span>")
+			return
+		user.visible_message("<span class='notice'>[user] begins engraving [src].</span>", \
+							 "<span class='notice'>You begin engraving [src].</span>")
+		if(!do_after(user, 40, target = src)) //This spits out a visible message that somebody is engraving a plaque, then has a delay.
+			return
+		name = "\improper [namechoice]" //We want improper here so examine doesn't get weird if somebody capitalizes the plaque title.
+		desc = "The plaque reads: '[descriptionchoice]'"
+		engraved = TRUE //The plaque now has a name, description, and can't be altered again.
+		user.visible_message("<span class='notice'>[user] engraves [src].</span>", \
+							 "<span class='notice'>You engrave [src].</span>")
+		return
+	if(istype(I, /obj/item/pen))
+		if(engraved)
+			to_chat(user, "<span class='warning'>This plaque has already been engraved, and your pen isn't fancy enough to engrave it anyway! Find a fountain pen.</span>")
+			return
+		to_chat(user, "<span class='warning'>Your pen isn't fancy enough to engrave this! Find a fountain pen.</span>") //Go steal the Curator's.
+		return
+	return ..()
+
+/obj/item/plaque/afterattack(atom/target, mob/user, proximity)
+	. = ..()
+	if(!iswallturf(target) || !proximity)
+		return
+	var/turf/T = target
+	var/turf/userT = get_turf(user)
+	var/obj/structure/plaque/S = new plaque_path(userT) //We place the plaque on the turf the user is standing, and pixel shift it to the target wall, as below.
+	//This is to mimic how signs and other wall objects are usually placed by mappers, and so they're only visible from one side of a wall.
+	var/dir = get_dir(userT, T)
+	switch(dir)
+		if(NORTH)
+			S.pixel_y = 32
+		if(SOUTH)
+			S.pixel_y = -32
+		if(EAST)
+			S.pixel_x = 32
+		if(WEST)
+			S.pixel_x = -32
+		if(NORTHEAST)
+			S.pixel_y = 32
+			S.pixel_x = 32
+		if(NORTHWEST)
+			S.pixel_y = 32
+			S.pixel_x = -32
+		if(SOUTHEAST)
+			S.pixel_y = -32
+			S.pixel_x = 32
+		if(SOUTHWEST)
+			S.pixel_y = -32
+			S.pixel_x = -32
```

Comment:
If there is a shorter, cleaner way to accomplish this, PLEASE let me know.

Reply:
```cp
if(dir & NORTH)
	S.pixel_y = 32
else if(dir & SOUTH)
	S.pixel_y = -32
if(dir & EAST)
	S.pixel_x = 32
else if(dir & WEST)
	S.pixel_x = -32
```

Reply:
The short version is still preferable, as the performance gain from the other is very minor, and this is not a hot proc.

Reply:
OK, I changed it.